packages/store/src/services/store/utils/bulk-operation-progress.ts

```diff
@@ -0,0 +1,144 @@
+  createColoredProgressBar,
+  createIndeterminateProgressBar,
+  clearLines,
+  createAnimatedDots,
+} from '@shopify/cli-kit/node/ui'
+import colors from '@shopify/cli-kit/node/colors'
+
+const POLLING_INTERVAL_MS = 300
+const DOT_ANIMATION_INTERVAL_MS = 80
+
+export function isOperationComplete(operation: {remoteOperationStatus?: string}): boolean {
+  return operation.remoteOperationStatus === 'completed' || operation.remoteOperationStatus === 'failed'
+}
+
+export function renderExportProgress(completedCount: number, dotCounter = 0): string {
+  const progressBar = createIndeterminateProgressBar()
+  const rightText = completedCount > 0 ? `${completedCount} ${colors.dim('items processed')}` : ''
+  const dots = createAnimatedDots(dotCounter)
+  const statusText = createRightAlignedText(`Exporting${dots}`, rightText)
+  return `${progressBar}\n\n${statusText}`
+}
+
+export function renderImportProgress(completedCount: number, totalCount: number, dotCounter = 0): string {
+  const percentage = totalCount > 0 ? Math.round((completedCount / totalCount) * 100) : 0
+  const progressBar = createColoredProgressBar(percentage)
+  const rightText = totalCount > 0 ? `${completedCount} ${colors.dim(`/ ${totalCount}`)}` : ''
+  const dots = createAnimatedDots(dotCounter)
+  const statusText = createRightAlignedText(`Importing${dots}`, rightText)
+  return `${progressBar}\n\n${statusText}`
+}
+
+function getLineCount(text: string): number {
+  return text.split('\n').length
+}
+
+async function handleOperationCompletion(
+  operation: BulkDataOperationByIdResponse,
+  getCompletionMessage: (status: 'completed' | 'failed', completedCount: number) => string,
+): Promise<void> {
+  const status = operation.organization.bulkData.operation.status
+  const firstOp = operation.organization.bulkData.operation.storeOperations?.[0]
+  const completedCount = firstOp?.completedObjectCount ?? 0
+  const message = getCompletionMessage(status === 'COMPLETED' ? 'completed' : 'failed', completedCount)
+  if (status === 'COMPLETED') {
+    outputCompleted(message)
+  } else {
+    outputWarn(message)
+  }
+}
+
+async function pollWithAnimation(
+  pollOperation: (operationId: string) => Promise<BulkDataOperationByIdResponse>,
+  operationId: string,
+  renderProgress: (operation: BulkDataOperationByIdResponse, dotCounter: number) => string,
+  updateProgress: (message: string, isProgressUpdate?: boolean) => void,
+): Promise<BulkDataOperationByIdResponse> {
+  let currentOperation: BulkDataOperationByIdResponse
+  let dotCounter = 0
+
+  const animationInterval = setInterval(() => {
+    if (
+      !currentOperation ||
+      currentOperation.organization.bulkData.operation.status === 'COMPLETED' ||
+      currentOperation.organization.bulkData.operation.status === 'FAILED'
+    )
+      return
+
+    const progressContent = renderProgress(currentOperation, dotCounter)
+    if (progressContent) {
+      updateProgress(progressContent, true)
+    }
+    dotCounter++
```

Comment:
nit, I wouldn't call it a `dotCounter` since it doesn't represent the actual count of dots. IMO something like `iteration` would be more clear. 

Not for this PR but I think we might get a better abstraction with separate class that keeps track of the counter and manages animation timer.